css/src/atomics/border.scss

```diff
@@ -0,0 +1,83 @@
+$border-sides: ('top', 'right', 'bottom', 'left');
+$border-positions: ('top', $user-right, 'bottom', $user-left);
+
+@mixin sides-rules($screen-size: '') {
```

Comment:
Personally, I like using sass objects for this type of thing. What are you thoughts on this?

```scss
$borders-directions: (
	'top': 'top',
	'left': $user-left,
	'bottom': 'bottom',
	'right': $user-right
);

$border-sides: 'top', 'right', 'bottom', 'left';
$border-positions: 'top', 'left', 'bottom', $user-left;
$border-width: 1px;

.border {
	border: $border-width solid $border !important;
}

.border-tablet {
	@include tablet {
		border: $border-width solid $border !important;
	}
}

.border-desktop {
	@include desktop {
		border: $border-width solid $border !important;
	}
}

.border-widescreen {
	@include widescreen {
		border: $border-width solid $border !important;
	}
}

@each $border-key, $direction in $borders-directions {
	.border-#{$border-key} {
		border-#{$direction}: $border-width solid $border !important;
	}
	.border-#{$border-key}-tablet {
		@include tablet {
			border-#{$direction}: $border-width solid $border !important;
		}
	}
	.border-#{$border-key}-desktop {
		@include desktop {
			border-#{$direction}: $border-width solid $border !important;
		}
	}
	.border-#{$border-key}-widescreen {
		@include widescreen {
			border-#{$direction}: $border-width solid $border !important;
		}
	}
}
```

Reply:
That approach definitely works as well, but adding the `-none` rules for each side and screen size which is missed currently in the example above, will make the code a bit messier, and the output is harder to read:

```
.border {
	border: 1px solid black !important;
}

.border-none {
	border: 0 !important;
}

@media screen and (min-width: 768px), print {
	.border-tablet {
		border: 1px solid black !important;
	}

	.border-none-tablet {
		border: 0 !important;
	}
}

@media screen and (min-width: 1088px) {
	.border-desktop {
		border: 1px solid black !important;
	}

	.border-none-desktop {
		border: 0 !important;
	}
}

@media screen and (min-width: 1472px) {
	.border-widescreen {
		border: 1px solid black !important;
	}

	.border-none-widescreen {
		border: 0 !important;
	}
}

.border-top {
	border-top: 1px solid black !important;
}

@media screen and (min-width: 768px), print {
	.border-top-tablet {
		border-top: 1px solid black !important;
	}

	.border-top-none-tablet {
		border: 0 !important;
	}
}

@media screen and (min-width: 1088px) {
	.border-top-desktop {
		border-top: 1px solid black !important;
	}

	.border-top-none-desktop {
		border: 0 !important;
	}
}

@media screen and (min-width: 1472px) {
	.border-top-widescreen {
		border-top: 1px solid black !important;
	}

	.border-top-none-widescreen {
		border: 0 !important;
	}
}

... so on for each side
```

Comparing to this output:
```
.border {
	border: 1px solid black !important;
}

.border-none {
	border: 0 !important;
}

.border-top {
	border-top: 1px solid black !important;
}

.border-top-none {
	border: 0 !important;
}

...

@media screen and (min-width: 768px), print {
	.border-none-tablet {
		border: 0 !important;
	}

	.border-top-tablet {
		border-top: 1px solid black !important;
	}

	.border-top-tablet-none {
		border: 0 !important;
	}

	...
}

@media screen and (min-width: 1088px) {
	.border-none-desktop {
		border: 0 !important;
	}

	.border-top-desktop {
		border-top: 1px solid black !important;
	}

	.border-top-desktop-none {
		border: 0 !important;
	}

	...
}

@media screen and (min-width: 1472px) {
	.border-none-widescreen {
		border: 0 !important;
	}

	.border-top-widescreen {
		border-top: 1px solid black !important;
	}

	...
}
```

I understand nobody would read the output, but I personally like when things are organized. And seeing a lot of media queries drives me crazy :D But let me know if you still think that I should update it anyways!

Reply:
Per outside discussion, we agreed we'd go with only tablet for now.

Reply:
updated!